Source/WebCore/html/ImageDataArray.cpp

```diff
@@ -184,6 +188,18 @@ Ref<JSON::Value> ImageDataArray::copyToJSONArray() const
     });
 }
 
+Ref<ArrayBufferView> ImageDataArray::extractBufferViewWithStorageFormat(std::optional<ImageDataStorageFormat> overridingStorageFormat) &&
+{
+    if (!overridingStorageFormat)
+        return WTFMove(m_arrayBufferView);
+
+    switch (*overridingStorageFormat) {
+    case ImageDataStorageFormat::Uint8: return asUint8ClampedArray();
+    case ImageDataStorageFormat::Float16: return asFloat16Array();
+    default: RELEASE_ASSERT_NOT_REACHED("Unexpected ImageDataStorageFormat value");
```

Comment:
Nit: slightly better to put the assert just below the `switch`, so that we can rely on the compiler telling us if there is a new, valid enum value that we haven't handled yet.